File: bubblesort.java
public class bubblesort{
	static void bubblesort(int[] a){
		int n = a.length;

		for(int i=0;i<n-1;i++){
			for(int j=0;j<n-i-1;j++){
				if(a[j]>a[j+1]){
					int temp = a[j];
					a[j] = a[j+1];
					a[j+1] = temp;
				}
			}
		}
	}
	public static void main(String[] args){
		int[] a = {7,6,4,8,2,1};
		bubblesort(a);
		for(int x:a){
			System.out.print(x+" ");
		}
		}
}

File: bubbleSort.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
void bubblesort(int arr[],int n){
        for(int i=0;i<n-1;i++){
                for(int j=0;j<n-i-1;j++){
                        if(arr[j]>arr[j+1]){
                                int temp = arr[j];
                                arr[j] = arr[j+1];
                                arr[j+1] = temp;
                        }
                }
        }
}
void generate_numbers(int arr[],int n){
	for(int i=0;i<n;i++){
		arr[i] = rand()%100000;
	}
}

int main(){
	srand(time(NULL));
	int size1 = 40000;
	int size2 = 80000;
	int*arr1 = (int*)malloc(size1*sizeof(int));
	int*arr2 = (int*) malloc(size2*sizeof(int));
	generate_numbers(arr1,size1);
	clock_t start1 = cloack();
	bubblesort(arr1,size1);
	clock_t end1 = clock();
	double time1 = ((double)end1-start1)CLOACKS_PER_SEC;
	generate_numbers(
